### PR TITLE
Add sysmem buffers perf tests

### PR DESCRIPTION
**Pull request by pjanevskiTT**
### Issue

#861 

### Description

Add tests for perf of reading/writing through sysmem buffers. Perf tests are testing zero copy transfer using DMA, mapping of the buffers, as well as reading/writing when you both have to map the buffer and read/write through it. Tests should represent the examples of how you can use the sysmem buffer.

### List of the changes

- Add test for zero copy transfer to tensix
- Add test for zero copy transfer to DRAM
- Add test for mapping the buffers
- Add test for rw with both mapping and DMA transfer

### Testing
CI

### API Changes
/

### CODE REVIEW

File: tests/microbenchmark/test_perf_dma.cpp [hunk 1]
```diff
@@ -113,3 +153,121 @@ TEST(TestPerf, DMADram) {
...
+        .num_host_mem_ch_per_mmio_device = 0,
+    });
+
+    const chip_id_t mmio_chip = *cluster->get_target_mmio_device_ids().begin();
+
+    SysmemManager* sysmem_manager = cluster->get_chip(mmio_chip)->get_sysmem_manager();
+
+    const uint32_t one_mb = 1 << 20;
+
+    void* mapping = mmap(nullptr, one_mb, PROT_READ | PROT_WRITE, MAP_ANONYMOUS | MAP_PRIVATE | MAP_POPULATE, -1, 0);
```
Comment (broskoTT) [inline tests/microbenchmark/test_perf_dma.cpp:196]: why mmap?
Could you just create a simple vector or array with fixed size? That would match our client code better, right?

Comment (broskoTT) [reply tests/microbenchmark/test_perf_dma.cpp:196]: same below

File: tests/microbenchmark/test_perf_dma.cpp [hunk 2]
```diff
@@ -113,3 +153,121 @@ TEST(TestPerf, DMADram) {
...
+
+    const chip_id_t mmio_chip = *cluster->get_target_mmio_device_ids().begin();
+
+    SysmemManager* sysmem_manager = cluster->get_chip(mmio_chip)->get_sysmem_manager();
+
+    const uint32_t one_mb = 1 << 20;
+
+    const CoreCoord tensix_core = cluster->get_soc_descriptor(mmio_chip).get_cores(CoreType::TENSIX)[0];
+    {
+        void* mapping =
```
Comment (broskoTT) [inline tests/microbenchmark/test_perf_dma.cpp:222]: Why is perf_sysmem_read_write bit used in this and previous test?

File: tests/microbenchmark/test_perf_dma.cpp (hunk 1 above)
Comment (pjanevskiTT) [reply tests/microbenchmark/test_perf_dma.cpp:196]: so far we support only page-aligned buffers, that is guaranteed by mmap, not by random allocation

File: tests/microbenchmark/test_perf_dma.cpp (hunk 2 above)
Comment (pjanevskiTT) [reply tests/microbenchmark/test_perf_dma.cpp:222]: because this test also maps the buffer, doesn't just do read/write
